mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala

```diff
@@ -219,8 +221,17 @@ class VizServlet extends ScalatraServlet {
       if (dictOpt.isDefined && viewRegion.end <= dictOpt.get.length) {
         val sampleIds: List[String] = params("sample").split(",").toList
         val alignments: RDD[AlignmentRecord] = VizReads.readsData.getRaw(viewRegion, sampleIds)
-        val array: Array[AlignmentRecord] = alignments.collect
-        Ok(write(array))
+
+        // convert to GA4GH avro and build a response
+        val gaReads: List[GAReadAlignment] = alignments.map(GA4GHConverter.toGAReadAlignment)
+          .collect
+          .toList
+        val readResponse = GASearchReadsResponse.newBuilder()
+          .setAlignments(gaReads)
+          .build()
+
+        // write response
+        Ok(readResponse)
```

Comment:
write takes any scala object and formats it into pretty json for js to interpret easily. Let me pull and try it out.

Best,

Alyssa Morrow
akmorrow@berkeley.edu <mailto:akmorrow@berkeley.edu>
414-254-6645






> On Jun 16, 2016, at 4:05 PM, Frank Austin Nothaft <notifications@github.com> wrote:
> 
> In mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala <https://github.com/akmorrow13/mango/pull/3#discussion_r67439508>:
> 
> > @@ -219,8 +221,17 @@ class VizServlet extends ScalatraServlet {
> >        if (dictOpt.isDefined && viewRegion.end <= dictOpt.get.length) {
> >          val sampleIds: List[String] = params("sample").split(",").toList
> >          val alignments: RDD[AlignmentRecord] = VizReads.readsData.getRaw(viewRegion, sampleIds)
> > -        val array: Array[AlignmentRecord] = alignments.collect
> > -        Ok(write(array))
> > +
> > +        // convert to GA4GH avro and build a response
> > +        val gaReads: List[GAReadAlignment] = alignments.map(GA4GHConverter.toGAReadAlignment)
> > +          .collect
> > +          .toList
> > +        val readResponse = GASearchReadsResponse.newBuilder()
> > +          .setAlignments(gaReads)
> > +          .build()
> > +
> > +        // write response
> > +        Ok(readResponse)
> What exactly does write(x) do? I originally had it as you wrote, but changed it because of the unittest failures I was getting over here.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/akmorrow13/mango/pull/3/files/db5ccff11512e5d2c012ed1c4dec28caf262ccad#r67439508>, or mute the thread <https://github.com/notifications/unsubscribe/AGM7-IkXhKo5AEHeBh6VG09X7ve-CcB1ks5qMdafgaJpZM4I34XS>.
>